pyblish_nuke/lib.py

```diff
@@ -255,6 +256,37 @@ def __init__(self):
 def dock(window):
     """ Expecting a window to parent into a Nuke panel, that is dockable. """
 
+    # Deleting existing dock
+    # There is a bug where existing docks are kept in-memory when closed via UI
+    if self._dock:
+        print "Deleting existing dock..."
+        self._dock.layout().addWidget(window)
+
+        parent = self._dock
+        dialog = None
+        stacked_widget = None
+        main_windows = []
+        # getting dock parents
+        while parent:
+            if parent.__class__.__name__ == "QDialog":
+                dialog = parent
+            if parent.__class__.__name__ == "QStackedWidget":
+                stacked_widget = parent
+            if parent.__class__.__name__ == "QMainWindow":
+                main_windows.append(parent)
+            parent = parent.parent()
+
+        dialog.deleteLater()
+
+        # if there are more than one main window, its a floating window
+        if len(main_windows) > 1:
+            # if the stacked widget only contains 1 widget, its empty
```

Comment:
Great descriptions. Without these I would have no idea what the code would mean or why you're doing this.

But how come you write comments in lowercase? :S